doc/man7/provider-signature.pod

```diff
@@ -22,17 +22,35 @@ provider-signature - The signature library E<lt>-E<gt> provider functions
  void OSSL_FUNC_signature_freectx(void *ctx);
  void *OSSL_FUNC_signature_dupctx(void *ctx);
 
+ /* Get the key types that a signature algorithm supports */
+ const char **OSSL_FUNC_signature_query_key_types(void);
+
  /* Signing */
  int OSSL_FUNC_signature_sign_init(void *ctx, void *provkey,
                                    const OSSL_PARAM params[]);
  int OSSL_FUNC_signature_sign(void *ctx, unsigned char *sig, size_t *siglen,
                               size_t sigsize, const unsigned char *tbs, size_t tbslen);
+ int OSSL_FUNC_signature_sign_message_init(void *ctx, void *provkey,
+                                           const OSSL_PARAM params[]);
+ int OSSL_FUNC_signature_sign_message_update(void *ctx, const unsigned char *in,
+                                             size_t inlen);
+ int OSSL_FUNC_signature_sign_message_final(void *ctx, unsigned char *sig,
+                                            size_t *siglen, size_t sigsize);
 
  /* Verifying */
  int OSSL_FUNC_signature_verify_init(void *ctx, void *provkey,
                                      const OSSL_PARAM params[]);
  int OSSL_FUNC_signature_verify(void *ctx, const unsigned char *sig, size_t siglen,
                                 const unsigned char *tbs, size_t tbslen);
+ int OSSL_FUNC_signature_verify_message_init(void *ctx, void *provkey,
+                                             const OSSL_PARAM params[]);
+ int OSSL_FUNC_signature_verify_message_update(void *ctx, const unsigned char *in,
+                                               size_t inlen);
+ /*
+  * OSSL_FUNC_signature_verify_final requires that the signature to be verified
+  * against is specified via an OSSL_PARAM.
```

Comment:
Which function is used to supply the `OSSL_PARAM`?

Reply:
Expanded that comment a bit